src/hooks/useLanguage.ts

```diff
@@ -1,14 +1,100 @@
 import { useCallback, useEffect, useState } from 'preact/hooks';
 import {
   ConsentManagerLanguageKey,
+  LanguageKey,
   TranslatedMessages,
   Translations,
 } from '@transcend-io/internationalization';
 import { settings } from '../settings';
 import { substituteHtml } from '../utils/substitute-html';
+import { invertSafe } from '@transcend-io/type-utils';
 
 export const loadedTranslations: Translations = Object.create(null);
 
+/**
+ * Mapping of browser locale to AWS base translation key
+ *
+ * TODO: https://transcend.height.app/T-39777
+ * This is here for a quick fix of our CM UI translation logic. Copied from the monorepo's extract_intl.ts
+ * This should be removed ASAP (it should be in the intl repo and imported, not in here or extract_intl.ts)
+ */
+export const TRANSLATE_LOCALE = {
+  [LanguageKey.EsEs]: 'es',
+  [LanguageKey.NlNl]: 'nl',
+  [LanguageKey.NlBe]: 'nl',
+  [LanguageKey.Es419]: 'es-MX',
+  // This is a hack to fix the fact that we don't have a LanguageKey -> BrowserLanguageKey mapping
+  'es-MX': 'es-MX',
+  [LanguageKey.ZhHk]: 'zh-TW',
+  [LanguageKey.AfZz]: 'af',
+  [LanguageKey.Ar]: 'ar',
+  [LanguageKey.En]: 'en',
+  [LanguageKey.Fr]: 'fr',
+  [LanguageKey.Es]: 'es',
+  [LanguageKey.De]: 'de',
+  [LanguageKey.It]: 'it',
+  [LanguageKey.Ja]: 'ja',
+  [LanguageKey.Ru]: 'ru',
+  [LanguageKey.ArAe]: 'ar',
+  [LanguageKey.FrFr]: 'fr',
+  [LanguageKey.DeDe]: 'de',
+  [LanguageKey.ItIt]: 'it',
+  [LanguageKey.BgBg]: 'bg',
+  [LanguageKey.ZhCn]: 'zh',
+  [LanguageKey.HrHr]: 'hr',
+  [LanguageKey.CsCz]: 'cs',
+  [LanguageKey.DaDk]: 'da',
+  [LanguageKey.EnGb]: 'en',
+  [LanguageKey.FiFi]: 'fi',
+  [LanguageKey.ElGr]: 'el',
+  [LanguageKey.HiIn]: 'hi',
+  [LanguageKey.HuHu]: 'hu',
+  [LanguageKey.IdId]: 'id',
+  [LanguageKey.JaJp]: 'ja',
+  [LanguageKey.KoKr]: 'ko',
+  [LanguageKey.LtLt]: 'lt',
+  [LanguageKey.MsMy]: 'ms',
+  [LanguageKey.NbNi]: 'no',
+  [LanguageKey.PlPl]: 'pl',
+  [LanguageKey.PtBr]: 'pt',
+  [LanguageKey.PtPt]: 'pt',
+  [LanguageKey.RoRo]: 'ro',
+  [LanguageKey.RuRu]: 'ru',
+  [LanguageKey.SrLatnRs]: 'sr',
+  [LanguageKey.SvSe]: 'sv',
+  [LanguageKey.TaIn]: 'ta',
+  [LanguageKey.ThTh]: 'th',
+  [LanguageKey.TrTr]: 'tr',
+  [LanguageKey.UkUa]: 'uk',
+  [LanguageKey.ViVn]: 'vi',
+  [LanguageKey.EnUS]: 'en',
+  [LanguageKey.EnAu]: 'en',
+  [LanguageKey.FrBe]: 'fr',
+  [LanguageKey.EnIe]: 'en',
+  [LanguageKey.EnCa]: 'en',
+  [LanguageKey.EnAe]: 'en',
+  [LanguageKey.DeAt]: 'de',
+  [LanguageKey.DeCh]: 'de',
+  [LanguageKey.ItCh]: 'it',
+  [LanguageKey.FrCh]: 'fr',
+  [LanguageKey.HeIl]: 'he',
+  [LanguageKey.EnNz]: 'en',
+  [LanguageKey.EtEe]: 'et',
+  [LanguageKey.IsIs]: 'is',
+  [LanguageKey.LvLv]: 'lv',
+  [LanguageKey.MtMt]: 'mt',
+  [LanguageKey.SkSk]: 'sk',
+  [LanguageKey.SlSl]: 'sl',
+  [LanguageKey.MrIn]: 'mr',
+  [LanguageKey.ZuZa]: 'en',
```

Comment:
Instead of embedding this table into our source how about simply splitting keys based on case boundaries (`/[a-z][A-Z]/`) and taking the first key chunk?

Reply:
Upon reviewing the LanguageKey enum it appears that every first chunk matches the first section of the language code

Reply:
unless im misunderstanding, that wouldnt work for dialects with aws translations (ie traditional chinese or mexican spanish)

Reply:
I mean for determining sub-languages for matching. I also noticed that this PR isn't using matchLanguages

Reply:
i dont think matchLanguages is quite what we want, we should probably delete it. ill make a note to do that on the refactor

my reasoning is that it considers the base tag supported if we support any subtag of it, which may technically be true in a theoretical sense, but it may not be supported in the sense of "can we serve a UI of it". if a user has their preferred language as `zh` - i.e. the most statistically likely choice for our users, lets say simplified chinese characters - and we only support traditional chinese characters - i.e. `zh-TW` - then we would mark simplified chinese characters as supported despite only being able to show traditional characters

you could make the argument that a `zh` preference means we should attempt to serve any subtag (simplified OR traditional characters) instead of its most popular dialect, but i would argue that its far more likely that a user e.g. selecting base German is just doing that because they reside in Germany and speak the Germany dialect of German and its the "base" language, not because they want to be served German German OR Swiss German OR Austrian German